qubes/tests/integ/basic.py

```diff
@@ -105,15 +105,42 @@ def test_120_start_standalone_with_cdrom_dom0(self):
         # check if VM do not crash instantly
         self.loop.run_until_complete(asyncio.sleep(5))
         self.assertTrue(self.vm.is_running())
-        # Type 'poweroff'
+        # Type 'halt'
         subprocess.check_call(['xdotool', 'search', '--name', self.vm.name,
-                               'type', '--window', '%1', 'poweroff\r'])
+                               'type', '--window', '%1', 'halt\r'])
         for _ in range(10):
             if not self.vm.is_running():
                 break
             self.loop.run_until_complete(asyncio.sleep(1))
         self.assertFalse(self.vm.is_running())
 
+    def test_121_start_uefi(self):
+        vmname = self.make_vm_name('appvm')
+        self.vm = self.app.add_new_vm('StandaloneVM', label='red', name=vmname)
+        self.loop.run_until_complete(self.vm.create_on_disk())
+        self.vm.kernel = None
+        self.vm.virt_mode = 'hvm'
+        subprocess.check_call(['qvm-features', '--', vmname, 'uefi', '1'])
```

Comment:
```suggestion
        self.vm.features["uefi"] = "1"
```
?

But also, have you tried if that works? Because the check_call here would deadlock...

Reply:
I have not tested this at all (beyond “it has no syntax errors”), because I don’t have a machine to test it on.